File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def digit_product(num):
            product = 1
            for digit in str(num):
                product *= int(digit)
            return product
        
        while True:
            if digit_product(n) % t == 0:
                return n
            n += 1

File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.js
/**
 * @param {number} n
 * @param {number} t
 * @return {number}
 */
var smallestNumber = function(n, t) {
    while (true) {
        let product = 1;
        for (const digit of String(n).split('')) {
            product *= digit;
        }
        if (product % t === 0) {
            return n;
        }
        else {
            n++;
        }
    }
};